include/boost/gil/extension/toolbox/color_spaces/lab.hpp

```diff
@@ -19,7 +19,7 @@
 ///
 ////////////////////////////////////////////////////////////////////////////////////////
 
-#include <boost/gil/gil_all.hpp>
+#include <boost/gil/gil_all.hpp> // FIXME: Include what you use, not everything, even in extensions!
```

Comment:
Fair enough. But meanwhile, we should replace `boost/gil/gil_all.hpp` by `boost/gil.hpp`, so we can retire `boost/gil/gil_all.hpp` ASAP.

Reply:
I considered that, but then I remembered this:

https://svn.boost.org/trac10/wiki/Guidelines/MaintenanceGuidelines
> Avoid the use of include all features files at the /boost level use instead specific files at boost/lib/file.hpp. Including less code reduce your user code breaking chances. 

So, I decided to keep the deprecated header.

Reply:
Well, OK, so the guideline is about including only the header that's needed. I don't disagree (though there are arguments for using a higher-level header, notably about robustness against code refactoring as we have just done.)
But my point really is that we are replacing `boost/gil/gil_all.hpp` by `boost/gil.hpp`, so if you are already touching that line above, you might as well remove the reference to a deprecated file.

Reply:
> But my point really is that we are replacing boost/gil/gil_all.hpp by boost/gil.hpp,
> so if you are already touching that line above, you might as well remove the reference to a deprecated file.

Since I don't know (or I'm too lazy now to find out) what detailed headers are required in this file,
I preferred to leave the deprecated header - if I switched to the preferred `gil.hpp`, chances are we will never properly refactor that issue what would be not good :)

Reply:
Hehe, that argument could have come from me. :-)
OK, let's keep it there as a reminder to come back and clean up at the latest when we are about to remove `gil/gil_all.hpp`.

Reply:
OK.

There is round of `#include` only refactoring - we need to switch from `"x.hpp" to qualified `<boost/gil/x.hpp>` as the former is discouraged.

Per https://www.boost.org/development/header.html
> Then both your code and user code specifies the sub-directory in #include directives.